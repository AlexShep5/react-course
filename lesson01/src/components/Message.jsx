import styles from './Message.module.scss'

export function Message(props) {

    return (
        <>
            <h1 className={styles.title}>{props.title}</h1>
        </>
    )
}
