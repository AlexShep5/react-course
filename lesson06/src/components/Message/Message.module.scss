.wrap {
    display: flex;
    color: green;
    padding: 8px 0;

    .author {
        padding-left: 16px;
        flex-grow: 1;
    }

    .text {
        padding-left: 16px;
        width: 70%;

    }
}
