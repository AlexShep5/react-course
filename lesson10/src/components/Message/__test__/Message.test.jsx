import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { Message } from '../Message'

describe('testing Message', () => {
  it('render component Message', () => {
    render(<Message></Message>)
  })

  it('render with snapshot', () => {
    const { asFragment } = render(<Message></Message>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('render with message', () => {
    render(
      <Message text='message'></Message>
    )
    expect(screen.getByText('message')).toBeInTheDocument()
  })

  it('render with author', () => {
    render(
      <Message author='author'></Message>
    )
    expect(screen.getByText('author')).toBeInTheDocument()
  })

  it('render multiply', () => {
    render(
      <>
        <Message text='message'></Message>
        <Message text='message'></Message>
      </>
    )
    expect(screen.queryAllByText('message').length).toBe(2)
  })

})
