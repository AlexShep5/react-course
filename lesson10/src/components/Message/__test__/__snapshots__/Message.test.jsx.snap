// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Message render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrap undefined"
  >
    <div
      class="author"
    />
    <div
      class="text"
    />
  </div>
</DocumentFragment>
`;
